src/Tasks/Microsoft.NET.Build.Tasks/WriteAppConfig.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using Microsoft.Build.Framework;
+using NuGet.Frameworks;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    public sealed class WriteAppConfig : TaskBase
+    {
+        /// <summary>
+        /// Path to the app.config source file.
+        /// </summary>
+        public ITaskItem AppConfigFile { get; set; }
+
+        [Required]
+        public string TargetFramework { get; set; }
```

Comment:
This task should be based on the `TargetFrameworkIdentifier` and `TargetFrameworkVersion` properties (which are by default inferred from `TargetFramework`) instead of directly depending on `TargetFramework`.

Reply:
used TargetFrameworkIdentifier and TargetFrameworkVersion instead